packages/react-router-dom/__tests__/special-characters-test.tsx

```diff
@@ -45,24 +54,24 @@ import {
 
 let specialChars = [
   // This set of characters never gets encoded by window.location
-  { char: "x", pathChar: "x", searchChar: "x", hashChar: "x" },
-  { char: "X", pathChar: "X", searchChar: "X", hashChar: "X" },
-  { char: "~", pathChar: "~", searchChar: "~", hashChar: "~" },
-  { char: "!", pathChar: "!", searchChar: "!", hashChar: "!" },
-  { char: "@", pathChar: "@", searchChar: "@", hashChar: "@" },
-  { char: "$", pathChar: "$", searchChar: "$", hashChar: "$" },
-  { char: "*", pathChar: "*", searchChar: "*", hashChar: "*" },
-  { char: "(", pathChar: "(", searchChar: "(", hashChar: "(" },
-  { char: ")", pathChar: ")", searchChar: ")", hashChar: ")" },
-  { char: "_", pathChar: "_", searchChar: "_", hashChar: "_" },
-  { char: "-", pathChar: "-", searchChar: "-", hashChar: "-" },
-  { char: "+", pathChar: "+", searchChar: "+", hashChar: "+" },
-  { char: "=", pathChar: "=", searchChar: "=", hashChar: "=" },
-  { char: "[", pathChar: "[", searchChar: "[", hashChar: "[" },
-  { char: "]", pathChar: "]", searchChar: "]", hashChar: "]" },
-  { char: ":", pathChar: ":", searchChar: ":", hashChar: ":" },
-  { char: ";", pathChar: ";", searchChar: ";", hashChar: ";" },
-  { char: ",", pathChar: ",", searchChar: ",", hashChar: "," },
+  { char: "x", pathChar: "x", searchChar: "x", hashChar: "x", decodedChar: "x" },
+  { char: "X", pathChar: "X", searchChar: "X", hashChar: "X", decodedChar: "X" },
+  { char: "~", pathChar: "~", searchChar: "~", hashChar: "~", decodedChar: "~" },
+  { char: "!", pathChar: "!", searchChar: "!", hashChar: "!", decodedChar: "!" },
+  { char: "@", pathChar: "@", searchChar: "@", hashChar: "@", decodedChar: "@" },
+  { char: "$", pathChar: "$", searchChar: "$", hashChar: "$", decodedChar: "$" },
+  { char: "*", pathChar: "*", searchChar: "*", hashChar: "*", decodedChar: "*" },
+  { char: "(", pathChar: "(", searchChar: "(", hashChar: "(", decodedChar: "(" },
+  { char: ")", pathChar: ")", searchChar: ")", hashChar: ")", decodedChar: ")" },
+  { char: "_", pathChar: "_", searchChar: "_", hashChar: "_", decodedChar: "_" },
+  { char: "-", pathChar: "-", searchChar: "-", hashChar: "-", decodedChar: "-" },
+  { char: "+", pathChar: "+", searchChar: "+", hashChar: "+", decodedChar: "+" },
+  { char: "=", pathChar: "=", searchChar: "=", hashChar: "=", decodedChar: "=" },
+  { char: "[", pathChar: "[", searchChar: "[", hashChar: "[", decodedChar: "[" },
+  { char: "]", pathChar: "]", searchChar: "]", hashChar: "]", decodedChar: "]" },
+  { char: ":", pathChar: ":", searchChar: ":", hashChar: ":", decodedChar: ":" },
+  { char: ";", pathChar: ";", searchChar: ";", hashChar: ";", decodedChar: ";" },
+  { char: ",", pathChar: ",", searchChar: ",", hashChar: ",", decodedChar: "," },
```

Comment:
Added the `decodedChar` to all existing setups - even though it's the same as the `char` since we now assert matched param values against `decodedChar`